clang/lib/AST/ByteCode/InterpBuiltin.cpp

```diff
@@ -2206,6 +2206,59 @@ static unsigned computePointerOffset(const ASTContext &ASTCtx,
   return Result;
 }
 
+/// __builtin_assume_dereferenceable(Ptr, Size)
+static bool interp__builtin_assume_dereferenceable(InterpState &S, CodePtr OpPC,
+                                                   const InterpFrame *Frame,
+                                                   const CallExpr *Call) {
+  assert(Call->getNumArgs() == 2);
+
+  APSInt ReqSize = popToAPSInt(S, Call->getArg(1));
+  const Pointer &Ptr = S.Stk.pop<Pointer>();
+
+  if (ReqSize.isZero())
+    return true;
+  if (Ptr.isZero()) {
+    S.FFDiag(S.Current->getSource(OpPC), diag::note_constexpr_access_null)
+        << AK_Read << S.Current->getRange(OpPC);
+    return false;
+  }
+  if (!Ptr.isBlockPointer()) {
+    if (Ptr.isIntegralPointer()) {
+      S.FFDiag(S.Current->getSource(OpPC), diag::note_constexpr_invalid_cast)
```

Comment:
That should be emitted elsewhere. If it isn't, just ignore it for now, but emitting this diagnostic here is wrong.